src/Chapter4.hs

```diff
@@ -611,6 +638,10 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+instance Monad List where
+  (>>=) :: List a -> (a -> List b) -> List b
+  (>>=) Empty _ = Empty
+  (>>=) (Cons x xs) f = concatList (f x) (xs >>= f)
```

Comment:
you can also just use `fmap` in here. As `fmap` already handles empty lists, you don't need to pattern match:

```haskell
xs >>= f = concatList (fmap f xs)
```

Reply:
Correct. I didn't think of this. Thank you 😊

Reply:
But here, concatList accepts two lists as input. I guess what you're suggesting here is to use a `flatten` method instead?

Reply:
Oh, yes, sorry, in the standard library, `concat` is for list flattering 😄